core/src/main/java/org/infinispan/remoting/rpc/RpcManager.java

```diff
@@ -148,4 +150,17 @@ void invokeRemotelyInFuture(NotifyingNotifiableFuture<Map<Address, Response>> fu
     */
    RpcOptions getDefaultRpcOptions(boolean sync, DeliverOrder deliverOrder);
 
+   static Map<Address, Response> waitFor(CompletableFuture<Map<Address, Response>> future, RpcOptions options)
+         throws InterruptedException, ExecutionException {
+      long waitTime = options.timeout() * 2;
```

Comment:
you have this exact same logic in https://github.com/infinispan/infinispan/pull/3841/files#diff-36bedadd971913e8bd8c08cc2e79e9a2R934 below. Maybe coalesce them ?